app/assets/src/components/views/SampleView/SampleView.jsx

```diff
@@ -299,7 +299,15 @@ class SampleView extends React.Component {
   renderSampleReportError = () => {
     const { pipelineRun, sample } = this.props;
     let status, message, linkText, issueType, link;
-    switch (pipelineRun.known_user_error) {
+    switch (pipelineRun.known_user_error || pipelineRun.error_message) {
```

Comment:
I believe field `error_message` should be used to store a description of the issue, and the error code (`BASESPACE_UPLOAD_FAILED`) should be stored in the `known_user_error`.

Reply:
👍

Reply:
That makes sense regarding `error_message`. But not sure about `known_user_error`.

In my mind, `known_user_error` means an error on the user's part, instead of a technical error. A basespace upload failing could be due to user error (giving us an invalid dataset ID...but this is unlikely since they are picking files in our upload flow, not giving us dataset IDs directly), but more likely it is a network error or maybe the Basespace API changed.

A new field like `error_type` might be more appropriate. `known_user_error` seems like it's the kind of error where we wouldn't send an alert in Datadog, but I do think we want to send it in the basespace_upload_failed case.

Thoughts? Should I just go ahead and use `known_user_error`?

Reply:
Makes sense, but I believe in this case we need to create support for non-user errors, like the error_type field you suggested (or perhaps error_code)

Reply:
I can do `error_code`. I'll add a comment that `error_code` is for non-user errors.

Reply:
Actually I prefer `error_type`, which implies a string. I think it's inconvenient to pass around a non-human-readable number, and then convert it to a string. For example, you'd have to convert it in the code above when you do the switch statement.